aiida_kkr/workflows/BandStructure_WF.py

```diff
@@ -0,0 +1,443 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+
+from six.moves import range
+from aiida.orm import Code,  Dict, RemoteData, StructureData, Float
+from aiida.orm import XyData, WorkChainNode, load_node, CalcJobNode, ArrayData, KpointsData
+from aiida.engine import WorkChain, if_, ToContext, submit
+from aiida.engine import CalcJob, calcfunction
+from masci_tools.io.kkr_params import kkrparams
+from aiida_kkr.tools.common_workfunctions import test_and_get_codenode, get_parent_paranode, update_params_wf, get_inputs_kkr
+from aiida_kkr.calculations.kkr import KkrCalculation
+from aiida_kkr.calculations.voro import VoronoiCalculation
+from aiida.common.exceptions import InputValidationError, ConfigurationError
+from aiida_kkr.tools.save_output_nodes import create_out_dict_node
+from aiida.tools.data.array.kpoints import get_explicit_kpoints_path
+__copyright__ = u"This is the Copyright"
+__license__ = "Here is the license"
+__version__ = "Here is the version"
+__contributors__ = "using technique from Philpp RÜssmann's DOS wf"
+
+
+
+class kkr_BS_wf(WorkChain):
+    """
+    Workchain for BandStructure calculation, starting from RemoteFolderData the previous converged KKR calculation remote folder data
+
+    inputs:
+    :param wf_parameters: (Dict), (optional); Workchain Specifications, contains npt2, tempr, emin(ev), emax(ev), kmesh
+                          keys. The Energy emin and emax are the energy difference from the fermi energy.
+
+    :param options: (Dict), (optional); Computer Specifications, Schedualer command, parallel or serial
+    :param kpoints: (KpointsData),(optional); Kpoints data type from thestructure,
+                                   but not mendatory as it can be extrruct from structure internaly from the remote data
+    :param remote_data: (RemoteData)(mendaory); From the previous kkr-converged calcualtion.
+    :param kkr: (Code)(mendaory); KKR code specifiaction
+    :param label: (Str): 
+    :param description: (Str): 
+    
+
+    returns:
+    :out BS_Data : (ArrayData) ; Consist of (Dos_intensity matrix), k_points (list), Energy_levels (list), special_kpoints(dict)
+    :out result_wf: (Dict); work_chain_specifications node, BS_data node, remote_folder node  
+    """
+
+    _wf_version = __version__
+    _wf_label = 'kkr_BandStructure_wc'
+    _wf_description = """Workflow for a bandstructure calculation starting eithe from a structure with automatic voronoi'
+                        calculation or a valid RemoteData of a previous calculation."""
+    _wf_params_default = {
+                        'EMIN':5,  # Energy below the fermi surface for energy contour in ev unit
+                        'EMAX': 5,     # Energy over the fermi surface for energy contour in ev unit
+                        "NPT2": 200,   # Energy points in the energy contour
+                        "kmesh": [30, 30, 30],
+                        'TEMPR': 50.   # temperature
```

Comment:
Why do the RCLUSTZ will have any impact on the  on the DOS calc? Because, it is already converged.

Reply:
for a qdos calculation you run  at much lower values of the imaginary part of the energies (given by NPOL=0 and a small value of the temperature). This puts higher demands on the numerics which in the end require larger screening clusters. Otherwise you will see negative densities.